### ISSUE TITLE
Unable to create the group "/sketches/": (Links) Object already exists

### ISSUE THREAD

**Issue by danielarothschild** [bacpop/PopPUNK#308]
**Versions**
Using PopPUNK version 2.6.5 and pp-sketchlib v2.1.2

**Command used and output returned**
Just trying to run the standard poppunk_assign using one of the downloaded databases from poppunk to cluster the genomes:
`poppunk_assign --db database --query qfile.txt --output poppunk_clusters --threads 8`

but this error comes up:
```
in constructDatabase
    pp_sketchlib.constructDatabase(db_name=dbname,
pp_sketchlib.HDF5Exception: Unable to create the group "/sketches/": (Links) Object already exists
```

I also tried to add the --overwrite option but nothing changed. I then tried re-installing poppunk and the database in a new conda environment (same error). So I tried to create my own database in case this was the issue using:
`poppunk --create-db --output database --r-files rlist.txt --threads 8`

but the same error came up.

I was able to run this before a few months ago with no issues. Sorry if it's basic, not an advanced programer here, any help is appreciated!

**Comment by johnlees** [comment]
Thanks for the clear description and versions

This is odd, and I think must be something with the installation. Is this installed with conda? What does `conda list` give?

Can you try with a different `--output` prefix too (with `--create-db`)?

Can you try with sketchlib directly? (see https://poppunk.bacpop.org/sketching.html#using-pp-sketchlib-directly)

**Comment by danielarothschild** [comment]
Thanks for your response. I'm still having issues.

Yes, I installed with conda. `conda list` gives the following:

`# packages in environment at /home/user/.conda/envs/pp_env:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
aom                       3.8.2                h59595ed_0    conda-forge
at-spi2-atk               2.38.0               h0630a04_3    conda-forge
at-spi2-core              2.40.3               h0630a04_0    conda-forge
atk-1.0                   2.38.0               hd4edc92_1    conda-forge
biopython                 1.83            py310h2372a71_0    conda-forge
brotli                    1.1.0                hd590300_1    conda-forge
brotli-bin                1.1.0                hd590300_1    conda-forge
brotli-python             1.1.0           py310hc6cd4ac_1    conda-forge
bzip2                     1.0.8                hd590300_5    conda-forge
c-ares                    1.27.0               hd590300_0    conda-forge
ca-certificates           2024.2.2             hbcca054_0    conda-forge
cached-property           1.5.2                hd8ed1ab_1    conda-forge
cached_property           1.5.2              pyha770c72_1    conda-forge
cairo                     1.18.0               h3faef2a_0    conda-forge
cairomm-1.16              1.16.2               h7e731d7_1    conda-forge
certifi                   2024.2.2           pyhd8ed1ab_0    conda-forge
cffi                      1.16.0          py310h2fee648_0    conda-forge
charset-normalizer        3.3.2              pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
contourpy                 1.2.0           py310hd41b1e2_0    conda-forge
cuda-version              12.4                 h3060b56_3    conda-forge
cycler                    0.12.1             pyhd8ed1ab_0    conda-forge
cython                    3.0.9           py310hc6cd4ac_0    conda-forge
dav1d                     1.2.1                hd590300_0    conda-forge
dbus                      1.13.6               h5008d03_3    conda-forge
dendropy                  4.6.1              pyhdfd78af_0    bioconda
docopt                    0.6.2                      py_1    conda-forge
epoxy                     1.5.10               h166bdaf_1    conda-forge
expat                     2.6.2                h59595ed_0    conda-forge
ffmpeg                    6.1.1           gpl_h38e077a_106    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 h77eed37_1    conda-forge
fontconfig                2.14.2               h14ed4e7_0    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
fonttools                 4.50.0          py310h2372a71_0    conda-forge
freetype                  2.12.1               h267a509_2    conda-forge
fribidi                   1.0.10               h36c2ea0_0    conda-forge
gdk-pixbuf                2.42.10              h829c605_5    conda-forge
gettext                   0.21.1               h27087fc_0    conda-forge
glib-tools                2.80.0               hde27a5a_1    conda-forge
gmp                       6.3.0                h59595ed_1    conda-forge
gnutls                    3.7.9                hb077bed_0    conda-forge
graph-tool                2.59            py310haee70ea_0    conda-forge
graph-tool-base           2.59            py310hca438cd_0    conda-forge
graphite2                 1.3.13            h58526e2_1001    conda-forge
gtk3                      3.24.41              h280cfa0_0    conda-forge
h5py                      3.10.0          nompi_py310h65828d5_101    conda-forge
harfbuzz                  8.3.0                h3d44ed6_0    conda-forge
hdbscan                   0.8.33          py310h1f7b6fc_4    conda-forge
hdf5                      1.14.3          nompi_h4f84152_100    conda-forge
hicolor-icon-theme        0.17                 ha770c72_2    conda-forge
icu                       73.2                 h59595ed_0    conda-forge
idna                      3.6                pyhd8ed1ab_0    conda-forge
joblib                    1.3.2              pyhd8ed1ab_0    conda-forge
keyutils                  1.6.1                h166bdaf_0    conda-forge
kiwisolver                1.4.5           py310hd41b1e2_1    conda-forge
krb5                      1.21.2               h659d440_0    conda-forge
lame                      3.100             h166bdaf_1003    conda-forge
lcms2                     2.16                 hb7c19ff_0    conda-forge
ld_impl_linux-64          2.40                 h41732ed_0    conda-forge
lerc                      4.0.0                h27087fc_0    conda-forge
libabseil                 20240116.1      cxx17_h59595ed_2    conda-forge
libaec                    1.1.3                h59595ed_0    conda-forge
libass                    0.17.1               h8fe9dca_1    conda-forge
libblas                   3.9.0           21_linux64_openblas    conda-forge
libboost                  1.82.0               h6fcfa73_6    conda-forge
libboost-python           1.82.0          py310hcb52e73_6    conda-forge
libbrotlicommon           1.1.0                hd590300_1    conda-forge
libbrotlidec              1.1.0                hd590300_1    conda-forge
libbrotlienc              1.1.0                hd590300_1    conda-forge
libcblas                  3.9.0           21_linux64_openblas    conda-forge
libcups                   2.3.3                h4637d8d_4    conda-forge
libcurl                   8.6.0                hca28451_0    conda-forge
libdeflate                1.19                 hd590300_0    conda-forge
libdrm                    2.4.120              hd590300_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 hd590300_2    conda-forge
libexpat                  2.6.2                h59595ed_0    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 13.2.0               h807b86a_5    conda-forge
libgfortran-ng            13.2.0               h69a702a_5    conda-forge
libgfortran5              13.2.0               ha4646dd_5    conda-forge
libgirepository           1.78.1               h003a4f0_1    conda-forge
libglib                   2.80.0               hf2295e7_1    conda-forge
libgomp                   13.2.0               h807b86a_5    conda-forge
libhwloc                  2.9.3           default_h554bfaf_1009    conda-forge
libiconv                  1.17                 hd590300_2    conda-forge
libidn2                   2.3.7                hd590300_0    conda-forge
libjpeg-turbo             3.0.0                hd590300_1    conda-forge
liblapack                 3.9.0           21_linux64_openblas    conda-forge
libnghttp2                1.58.0               h47da74e_1    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libopenblas               0.3.26          pthreads_h413a1c8_0    conda-forge
libopenvino               2024.0.0             h2e90f83_4    conda-forge
libopenvino-auto-batch-plugin 2024.0.0             hd5fc58b_4    conda-forge
libopenvino-auto-plugin   2024.0.0             hd5fc58b_4    conda-forge
libopenvino-hetero-plugin 2024.0.0             h3ecfda7_4    conda-forge
libopenvino-intel-cpu-plugin 2024.0.0             h2e90f83_4    conda-forge
libopenvino-intel-gpu-plugin 2024.0.0             h2e90f83_4    conda-forge
libopenvino-ir-frontend   2024.0.0             h3ecfda7_4    conda-forge
libopenvino-onnx-frontend 2024.0.0             h757c851_4    conda-forge
libopenvino-paddle-frontend 2024.0.0             h757c851_4    conda-forge
libopenvino-pytorch-frontend 2024.0.0             h59595ed_4    conda-forge
libopenvino-tensorflow-frontend 2024.0.0             hca94c1a_4    conda-forge
libopenvino-tensorflow-lite-frontend 2024.0.0             h59595ed_4    conda-forge
libopus                   1.3.1                h7f98852_1    conda-forge
libpciaccess              0.18                 hd590300_0    conda-forge
libpng                    1.6.43               h2797004_0    conda-forge
libprotobuf               4.25.3               h08a7969_0    conda-forge
librsvg                   2.56.3               he3f83f7_1    conda-forge
libsqlite                 3.45.2               h2797004_0    conda-forge
libssh2                   1.11.0               h0841786_0    conda-forge
libstdcxx-ng              13.2.0               h7e041cc_5    conda-forge
libtasn1                  4.19.0               h166bdaf_0    conda-forge
libtiff                   4.6.0                ha9c0a0a_2    conda-forge
libunistring              0.9.10               h7f98852_0    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libva                     2.21.0               hd590300_0    conda-forge
libvpx                    1.14.0               h59595ed_0    conda-forge
libwebp-base              1.3.2                hd590300_0    conda-forge
libxcb                    1.15                 h0b41bf4_0    conda-forge
libxcrypt                 4.4.36               hd590300_1    conda-forge
libxkbcommon              1.6.0                hd429924_1    conda-forge
libxml2                   2.12.6               h232c23b_0    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
mandrake                  1.2.2           py310h24473af_7    conda-forge
matplotlib-base           3.8.3           py310h62c0568_0    conda-forge
munkres                   1.0.7                      py_1    bioconda
ncurses                   6.4.20240210         h59595ed_0    conda-forge
nettle                    3.9.1                h7ab15ed_0    conda-forge
networkx                  3.2.1              pyhd8ed1ab_0    conda-forge
numpy                     1.26.4          py310hb13e2d6_0    conda-forge
ocl-icd                   2.3.2                hd590300_1    conda-forge
openblas                  0.3.26          pthreads_h7a3da1a_0    conda-forge
openh264                  2.4.1                h59595ed_0    conda-forge
openjpeg                  2.5.2                h488ebb8_0    conda-forge
openssl                   3.2.1                hd590300_1    conda-forge
p11-kit                   0.24.1               hc5aa10d_0    conda-forge
packaging                 24.0               pyhd8ed1ab_0    conda-forge
pandas                    2.2.1           py310hcc13569_0    conda-forge
pango                     1.52.1               ha41ecd1_0    conda-forge
pcre2                     10.43                hcad00b1_0    conda-forge
pillow                    10.2.0          py310h01dd4db_0    conda-forge
pip                       24.0               pyhd8ed1ab_0    conda-forge
pixman                    0.43.2               h59595ed_0    conda-forge
plotly                    5.19.0             pyhd8ed1ab_0    conda-forge
poppunk                   2.6.5           py310heb72de9_0    bioconda
pp-sketchlib              2.1.2           py310h37665e0_1    conda-forge
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
pugixml                   1.14                 h59595ed_0    conda-forge
pycairo                   1.26.0          py310hda9f760_0    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pygobject                 3.48.1          py310h30b043a_0    conda-forge
pyparsing                 3.1.2              pyhd8ed1ab_0    conda-forge
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
python                    3.10.14         hd12c33a_0_cpython    conda-forge
python-dateutil           2.9.0              pyhd8ed1ab_0    conda-forge
python-tzdata             2024.1             pyhd8ed1ab_0    conda-forge
python_abi                3.10                    4_cp310    conda-forge
pytz                      2024.1             pyhd8ed1ab_0    conda-forge
rapidnj                   2.3.2                h4ac6f70_4    bioconda
readline                  8.2                  h8228510_1    conda-forge
requests                  2.31.0             pyhd8ed1ab_0    conda-forge
scikit-learn              1.4.1.post1     py310h1fdf081_0    conda-forge
scipy                     1.12.0          py310hb13e2d6_2    conda-forge
setuptools                69.2.0             pyhd8ed1ab_0    conda-forge
sigcpp-3.0                3.6.0                h59595ed_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
snappy                    1.1.10               h9fff704_0    conda-forge
sparsehash                2.0.2                         0    bioconda
svt-av1                   1.8.0                h59595ed_0    conda-forge
tbb                       2021.11.0            h00ab1b0_1    conda-forge
tenacity                  8.2.3              pyhd8ed1ab_0    conda-forge
threadpoolctl             3.4.0              pyhc1e730c_0    conda-forge
tk                        8.6.13          noxft_h4845f30_101    conda-forge
tqdm                      4.66.2             pyhd8ed1ab_0    conda-forge
treeswift                 1.1.40             pyh7cba7a3_0    bioconda
tzdata                    2024a                h0c530f3_0    conda-forge
unicodedata2              15.1.0          py310h2372a71_0    conda-forge
urllib3                   2.2.1              pyhd8ed1ab_0    conda-forge
wayland                   1.22.0               h8c25dac_1    conda-forge
wheel                     0.43.0             pyhd8ed1ab_0    conda-forge
x264                      1!164.3095           h166bdaf_2    conda-forge
x265                      3.5                  h924138e_3    conda-forge
xkeyboard-config          2.41                 hd590300_0    conda-forge
xorg-compositeproto       0.4.2             h7f98852_1001    conda-forge
xorg-damageproto          1.2.1             h7f98852_1002    conda-forge
xorg-fixesproto           5.0               h7f98852_1002    conda-forge
xorg-inputproto           2.3.2             h7f98852_1002    conda-forge
xorg-kbproto              1.0.7             h7f98852_1002    conda-forge
xorg-libice               1.1.1                hd590300_0    conda-forge
xorg-libsm                1.2.4                h7391055_0    conda-forge
xorg-libx11               1.8.7                h8ee46fc_0    conda-forge
xorg-libxau               1.0.11               hd590300_0    conda-forge
xorg-libxaw               1.0.14               h7f98852_1    conda-forge
xorg-libxcomposite        0.4.6                h0b41bf4_1    conda-forge
xorg-libxcursor           1.2.0                h0b41bf4_1    conda-forge
xorg-libxdamage           1.1.5                h7f98852_1    conda-forge
xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge
xorg-libxext              1.3.4                h0b41bf4_2    conda-forge
xorg-libxfixes            5.0.3             h7f98852_1004    conda-forge
xorg-libxi                1.7.10               h7f98852_0    conda-forge
xorg-libxinerama          1.1.5                h27087fc_0    conda-forge
xorg-libxmu               1.1.3                h7f98852_0    conda-forge
xorg-libxpm               3.5.17               hd590300_0    conda-forge
xorg-libxrandr            1.5.2                h7f98852_1    conda-forge
xorg-libxrender           0.9.11               hd590300_0    conda-forge
xorg-libxt                1.3.0                hd590300_1    conda-forge
xorg-libxtst              1.2.3             h7f98852_1002    conda-forge
xorg-randrproto           1.5.0             h7f98852_1001    conda-forge
xorg-recordproto          1.14.2            h7f98852_1002    conda-forge
xorg-renderproto          0.11.1            h7f98852_1002    conda-forge
xorg-util-macros          1.19.3               h7f98852_0    conda-forge
xorg-xextproto            7.3.0             h0b41bf4_1003    conda-forge
xorg-xproto               7.0.31            h7f98852_1007    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
zlib                      1.2.13               hd590300_5    conda-forge
zstandard                 0.22.0          py310h1275a96_0    conda-forge
zstd                      1.5.5                hfc55251_0    conda-forge`

I did try running with a different `--output` prefix as you suggested, it gave the same error.

I also tried sketchlib directly and I get  the same error after entering the first command `sketchlib sketch -l ./genomes/file_names_and_paths.txt -o kp_db -s 10000 -k 15,31,2 --cpus 8` 

error:
`pp_sketchlib.constructDatabase(db_name=args['-o'],
pp_sketchlib.HDF5Exception: Unable to create the group "/sketches/": (Links) Object already exists`

even though I deleted my previous poppunk installation and re-installed with conda. 

What I don't understand is why this is happening now because before I had zero issues running the same pipeline before, I'm only trying to do it again with the same strains as before plus a few more - could this have anything to do with this (although I did reinstall poppunk in a different environment)? Even so, I tried running `poppunk_assign` with `--update-db` and with `--overwrite` but the error is the same: `pp_sketchlib.constructDatabase(db_name=args['-o'], pp_sketchlib.HDF5Exception: Unable to create the group "/sketches/": (Links) Object already exists` 

sorry for the trouble! Thanks for the help.

**Comment by danielarothschild** [comment]
Thanks for the reply. 

Sorry I was having issues again, but finally found the cause of the error (path to sequences on the rlist.txt file was shortcutted and not complete...silly error) just fixed that and now it works smoothly. Thanks again!!
